keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/DoubleVertexSamples.java

```diff
@@ -39,4 +39,7 @@ public DoubleTensor getVariances() {
             .divInPlace(samples.size()-1);
     }
 
+    public DoubleTensor asTensor() {
+        return DoubleTensor.concat(samples.toArray(new DoubleTensor[0]));
```

Comment:
Actually, Intellij complained if I used a presized array 😆 

> In older Java versions using pre-sized array was recommended, as the reflection call which is necessary to create an array of proper size was quite slow. However since late updates of OpenJDK 6 this call was intrinsified, making the performance of the empty array version the same and sometimes even better, compared to the pre-sized version. Also passing pre-sized array is dangerous for a concurrent or synchronized collection as a data race is possible between the size and toArray call which may result in extra nulls at the end of the array, if the collection was concurrently shrunk during the operation.

>

Reply:
Oh nice, I did not know this!